src/components/messenger/lib/utils.tsx

```diff
@@ -39,15 +38,13 @@ export const highlightFilter = (text, filter) => {
   return text;
 };
 
-export function getUserHandle(primaryZID: string, wallets: Wallet[]) {
+export function getUserHandle(primaryZID: string, primaryWallet: string) {
```

Comment:
Note: I have updated `getUserHandle` to just take in a single wallet address rather than an array of wallets. As a result, I have updated whats being passed in to `getUserHandle` in other areas it was being used.

Reply:
a side question but how do we compute the `primaryWallet`? Is it simply the first wallet address in the list of `wallets:[]`? 🤔

Reply:
Yeah first wallet in list. We've never added any logic to determine which wallet would be the one to display (set as `primaryWallet`. We should really copy how we set a default primaryZID for the user on registration, then we'll allow the user to set a default if they wish to do so (like in your edit profile work). The user should be able to do the same with wallets. If that makes sense ?

Reply:
Doesn't the database have a flag that says if it's the user's primary wallet?

Reply:
oh yes there is! Okay nice, maybe we should just include that in the search results instead then

Reply:
makes sense